### PR TITLE
examples/ct: export more info from CT handler

### PR DESCRIPTION
**Pull request by daviddrysdale**
Move handler registration up to main(), and drop
RegisterHandlers().

### CODE REVIEW

File: examples/ct/handlers.go [hunk 1]
```diff
@@ -182,22 +186,44 @@ func NewLogContext(logID int64, prefix string, trustedRoots *PEMCertPool, rpcCli
...
+// to handle those entrypoints.
+func (c LogContext) Handlers(prefix string) PathHandlers {
+	if !strings.HasPrefix(prefix, "/") {
+		prefix = "/" + prefix
+	}
+	prefix = strings.TrimRight(prefix, "/")
+
+	// Bind the LogContext instance to give an appHandler instance for each entrypoint.
+	handlers := map[string]AppHandler{
+		prefix + ct.AddChainPath:          AppHandler{Context: c, Handler: addChain, Name: "AddChain", Method: http.MethodPost},
```
Comment (codingllama) [inline examples/ct/handlers.go]: nit: Make the method names constants and use them both here and at Entrypoints.

File: examples/ct/handlers.go [hunk 2]
```diff
@@ -182,22 +186,44 @@ func NewLogContext(logID int64, prefix string, trustedRoots *PEMCertPool, rpcCli
 	return ctx
 }
 
+// Handlers returns a map from URL paths (with the given prefix) and AppHandler instances
+// to handle those entrypoints.
+func (c LogContext) Handlers(prefix string) PathHandlers {
```
Comment (codingllama) [inline examples/ct/handlers.go]: Add a test for Handlers?

I don't think we need to check the entirety of the map (or it'll begin to look like a change detector test), but testing the prefix logic and checking that all methods from Entrypoints are represented seems reasonable.

File: examples/ct/handlers.go [hunk 3]
```diff
@@ -182,22 +186,44 @@ func NewLogContext(logID int64, prefix string, trustedRoots *PEMCertPool, rpcCli
...
+	if !strings.HasPrefix(prefix, "/") {
+		prefix = "/" + prefix
+	}
+	prefix = strings.TrimRight(prefix, "/")
+
+	// Bind the LogContext instance to give an appHandler instance for each entrypoint.
+	handlers := map[string]AppHandler{
```
Comment (codingllama) [inline examples/ct/handlers.go]: Suggestion: Return here?

return map[string]AppHandler{ ... }

File: examples/ct/handlers.go (hunk 2 above)
Comment (daviddrysdale) [inline examples/ct/handlers.go]: Done.

File: examples/ct/handlers.go (hunk 3 above)
Comment (daviddrysdale) [inline examples/ct/handlers.go]: Done (was left over from an earlier iteration that returned a `*PathHandler`)

File: examples/ct/handlers.go (hunk 1 above)
Comment (daviddrysdale) [inline examples/ct/handlers.go]: Done.

**Pull request merged**